### DISCUSSION TITLE
Panache Reactive - How to persist multiple entities?

### DISCUSSION THREAD

**Discussion by studioitc** [quarkusio/quarkus#24124] [Community]
Hello,
which is the best way to persist a list of entities within Panache.withTransaction?
I'm trying to persist a list of string coming from a file.

`List<Uni<MyEntity>> list = new ArrayList<>();`
`try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {`
 `String line;`
&nbsp;&nbsp;`while ((line = br.readLine()) != null) {`
&nbsp;&nbsp;&nbsp;&nbsp;` list.add(new MyEntity(null, line).persist());`
&nbsp;&nbsp;`    }`
`}`

Now i suppose i need to combine somehow the list, i've tried with

`Uni.combine().all().unis(list).combinedWith(v -> null).onItem()`

but nothing is stored on DB.

Any suggestions?

**Comment by andrekullmann** [comment]
Hi @studioitc,
I'm facing a very similar issue. Did you found a solution?

**Comment by joao-p-soares-alb** [comment]
same problem here. Did you guys found a solution?

**Comment by floka94** [comment]
Any news on this?

**Reply by joao-p-soares-alb** [reply]
```
Panache.withTransaction(() ->
        Uni.combine().all().unis(
                    session.persistAll(entityAList().toArray()),
                    session.persistAll(entityBList().toArray())
         ).asTuple().chain(unused -> Uni.createFrom().voidItem())
)
```

worked for me

**Comment by codelpap** [comment]
Hi there. I've started a similar thread [https://stackoverflow.com/questions/78101824/persist-multiple-entities-with-reactive-quarkus-and-hibernate-panache](https://stackoverflow.com/questions/78101824/persist-multiple-entities-with-reactive-quarkus-and-hibernate-panache) . I would appreciate if anyone with a solution could reply and share to the community. Thank you very much.
